useradmin/md5-lib.pl

```diff
@@ -205,14 +205,39 @@ sub check_sha512
 }
 
 # encrypt_sha512(password, [salt])
-# Hashes a password, possibly with the give salt, with SHA512
+# Hashes a password, possibly with the given salt, with SHA512
 sub encrypt_sha512
 {
 my ($passwd, $salt) = @_;
 $salt ||= '$6$'.substr(time(), -8).'$';
 return crypt($passwd, $salt);
 }
 
+# unix_crypt_supports_yescrypt()
+# Returns 1 if the built-in crypt() function can already do yescrypt
+sub unix_crypt_supports_yescrypt
+{
+my $hash = '$y$j9T$waHytoaqP/CEnKFroGn0S/$fxd5mVc2mBPUc3vv.cpqDckpwrWTyIm2iD4JfnVBi26';
+my $newhash = eval { crypt('test', $hash) };
```

Comment:
Curious, does this work on many systems?

Reply:
It's meant to work on Ubuntu 22.04 and Debian 11. I didn't expect it to work on Ubuntu 20.04 but it actually did, however at the same time, it failed on Debian 10. Unexpected.

Reply:
Yeah, the crypt() function doesn't always support all hashing types. That's why elsewhere in this file we fall back to using a Perl module if it's installed.